Emby.Server.Implementations/Library/LiveStreamHelper.cs

```diff
@@ -48,20 +48,29 @@ public async Task AddMediaInfoWithProbe(MediaSourceInfo mediaSource, bool isAudi
 
             if (!string.IsNullOrEmpty(cacheKey))
             {
-                FileStream jsonStream = AsyncFile.OpenRead(cacheFilePath);
                 try
                 {
-                    mediaInfo = await JsonSerializer.DeserializeAsync<MediaInfo>(jsonStream, _jsonOptions, cancellationToken).ConfigureAwait(false);
+                    FileStream jsonStream = AsyncFile.OpenRead(cacheFilePath);
 
-                    // _logger.LogDebug("Found cached media info");
+                    try
+                    {
+                        mediaInfo = await JsonSerializer.DeserializeAsync<MediaInfo>(jsonStream, _jsonOptions, cancellationToken).ConfigureAwait(false);
+                        // _logger.LogDebug("Found cached media info");
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogError(ex, "Error deserializing mediainfo cache");
+                    }
+
+                    await jsonStream.DisposeAsync().ConfigureAwait(false);
                 }
-                catch (Exception ex)
+                catch (IOException)
                 {
-                    _logger.LogError(ex, "Error deserializing mediainfo cache");
+                    _logger.LogDebug("Could not open cached media info");
```

Comment:
You should log the exception. IOException is very generic

Reply:
Updated :+1: